lib/port_scanner.py

```diff
@@ -15,25 +15,23 @@ def scan_port(host, port):
         sock.close()
 
 
-def scan_thread(host):
-    global queue
-    # TODO reimplement this loop with multiple ranges and threads
-    while True:
-        port_number = queue.get()
-        scan_port(host, port_number)
-        queue.task_done()
+def scan_thread(host, thread_start, thread_end):
+    for port in range(thread_start, thread_end + 1):
+        scan_port(host, port)
 
 
-def start_port_scan(host, ports, threads):
-    global queue
-    for thread in range(threads):
-        thread = Thread(target=scan_thread, args=(host,))
-        # if daemon is True that thread will end when the main threadends
-        thread.daemon = True
+def start_port_scan(host, ports_range, n_threads):
+    start, end = ports_range[0], ports_range[-1]    
+    step = (end - start + 1) // n_threads
+
+    threads = []
+    for i in range(n_threads):
+        thread_start = start + i * step
+        thread_end = start + (i + 1) * step - 1 if i != n_threads - 1 else end
```

Comment:
I believe the `-p` ranges should be inclusive, so entering range like: `-p 1-10` should scan ports: `1, 2, 3, 4, 5, 6, 7, 8, 9, 10`. If that's how he want to handle this, this code has a off by one error at the end. Scan 1-10, thread ranges:
```
thread start: 1 thread end: 2
thread start: 3 thread end: 4
thread start: 5 thread end: 6
thread start: 7 thread end: 9
```

Reply:
I printed results of scan_thread() and I'm sure port 10 is accessed
![image](https://github.com/PixelToolkit/PixelToolkit/assets/113286903/dba2adb5-c81b-4fdf-8503-21f735d49b9b)

Reply:
how many threads do you have in your computer?

Reply:
I was inputting manually threads in range 1 to 16 and always got 10 ports scanned in this function. Personally I have 8

Reply:
Ok, now that you changed to accepting `port_start` and `port_end` in the `scan_port_range` function this doesn't seem to be a problem anymore.